wallet/src/repository/wallet/mod.rs

```diff
@@ -496,6 +504,24 @@ where
                 _ => continue,
             };
 
+            // Check if tally txn corresponds to a wallet sent data request
+            if let types::Transaction::Tally(tally) = &txn {
+                // There is a DR transaction persisted in database whose tally was found
+                if self
+                    .db
+                    .get::<_, model::Path>(&keys::transactions_index(tally.dr_pointer.as_ref()))
+                    .is_ok()
+                {
+                    filtered_txns.push(txn.clone());
+                    continue;
+                }
+                // There is a DR transaction in pending state whose tally was found
+                else if let Some(_dr_tx) = state.pending_dr_movements.get(&tally.dr_pointer) {
```

Comment:
But here we do have a contains:

```suggestion
                else if state.pending_dr_movements.contains(&tally.dr_pointer) {
```

Reply:
we do not have a contains